### PR TITLE
[RAPTOR-10167] update envs' base images and drum

### PR DESCRIPTION
**Pull request by yakov-g**
# This repository is public. Do not put here any private DataRobot or customer's data: code, datasets, model artifacts, .etc.

## Summary
* update envs' base images with DRUM==v1.10.14 and MLOps==9.2.8
* update all the envs with DRUM 1.10.15
* For the torch env install torch package supporting CPU only, it saves 3-4Gb of space

### CODE REVIEW

**Comment by yakov-g** [comment]
jenkins run test_functional_general

**Comment by yakov-g** [comment]
jenkins run test_e2e_drop_in_envs

**Comment by yakov-g** [comment]
jenkins run test_functional_general

**Comment by yakov-g** [comment]
jenkins run test_e2e_drop_in_envs

File: docker/dropin_env_base_jdk/build_image.sh [hunk 1]
```diff
@@ -8,13 +8,13 @@
 
 SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
 
-IMAGE_NAME=datarobot/dropin-env-base-jdk
-IMAGE_TAG=debian11-py3.9-jdk11.0.16-drum1.10.7-mlops9.1.3
+IMAGE_NAME=datarobotdev/dropin-env-base-jdk
```
Comment (zohar-mizrahi) [inline docker/dropin_env_base_jdk/build_image.sh:11]: Is the `datarobotdev` name by intention?

File: public_dropin_environments/java_codegen/dr_requirements.txt [hunk 2]
```diff
@@ -1,3 +1,3 @@
 pandas<=2.0.3
 pyarrow>=0.14.1,<=14.0.1
-datarobot-drum==1.10.13
+datarobot-drum==1.10.15
```
Comment (zohar-mizrahi) [inline public_dropin_environments/java_codegen/dr_requirements.txt:3]: I noticed that until here you referred to `datarobot-drum==1.10.14`.
Then looking throughout your changes I notice that sometimes you use `1.10.14` and sometimes `1.10.15`.
Is it wrong?

File: docker/dropin_env_base_jdk/build_image.sh (hunk 1 above)
Comment (yakov-g) [reply docker/dropin_env_base_jdk/build_image.sh:11]: Yes.
They change structure, that images will be first uploaded to `datarobotdev`, and then promoted to `datarobot` by accounts with permissions.

File: public_dropin_environments/java_codegen/dr_requirements.txt (hunk 2 above)
Comment (yakov-g) [reply public_dropin_environments/java_codegen/dr_requirements.txt:3]: I worked on base images a couple days ago, when 1.10.15 was not yet released. So base images have drum 1.10.14.

And environments will have 1.10.15 now.

**Review by zohar-mizrahi — APPROVED** [review]

**Review by mxpoliakov — APPROVED** [review]

**Review by eric-s-s — APPROVED** [review]

**Pull request merged**